src/app/time-sheet/components/time-entry-dialogue/time-entry-dialogue.component.ts

```diff
@@ -7,87 +7,131 @@ import * as moment from 'moment';
-    @Inject(MAT_DIALOG_DATA) public data: moment.Moment,
+    @Inject(MAT_DIALOG_DATA) public data: any,
     private userService: UserService,
     private timeEntryDialogueService: TimeEntryDialogueService
   ) { }
 
   ngOnInit(): void {
-    this.currentDate = this.data;
-    this.getCurrentUser();
-    this.getUserProjects();
+    this.currentDate = this.data.date;
+    this.entryId = this.data.EntryId;
     this.TimeEntryForm = new FormGroup({
-      project: new FormControl(null, [Validators.required]),
-      task: new FormControl({ value: null, disabled: true }, [
+      projectID: new FormControl(null, [Validators.required]),
+      taskID: new FormControl({ value: null, disabled: true }, [
         Validators.required
       ]),
       hours: new FormControl(null, [Validators.required]),
       minutes: new FormControl(null, [Validators.required]),
       notes: new FormControl(null)
     });
 
+    this.observables.push(this.userService.getCurrentUser());
+    this.observables.push(this.timeEntryDialogueService.getProjectsForCurrentUser());
+    if (this.isEdit) {
+      this.observables.push(this.timeEntryDialogueService.getTaskEntry(this.entryId));
+    }
+
+    forkJoin(this.observables).subscribe(
+      responseList => {
+        this.currentUser = responseList[0] as User;
+        this.userProjects = responseList[1] as UserProject[];
+        this.editEntry = responseList[2] as TaskEntryUpdate;
+        if (this.isEdit) {
+          this.displayEntry();
+        }
+        this.TimeEntryForm.get('projectID').valueChanges.subscribe(val => {
+          this.toggleTaskDropdown(val);
+        });
+      }
+    );
   }
 
   get f(): any {
     return this.TimeEntryForm.controls;
   }
 
+  get isEdit(): boolean {
+    return this.entryId ? true : false;
+  }
+
   closeDialog(): void {
     this.dialogRef.close();
   }
 
-  toggleTaskDropdown() {
-    this.f.task.reset();
-    if (this.f.task.status === 'DISABLED') {
-      this.f.task.enable();
+  toggleTaskDropdown(selectedProjectId: number) {
+    this.f.taskID.reset();
+    if (this.f.taskID.status === 'INVALID' || this.f.taskID.status === 'DISABLED') {
+      this.f.taskID.enable();
     } else {
-      this.f.task.disable();
+      this.f.taskID.disable();
     }
+    this.projectTasks = this.userProjects.find(x => x.projectID === selectedProjectId).tasks;
   }
 
   onSubmit() {
     if (this.TimeEntryForm.invalid || this.TimeEntryForm.untouched) {
       return;
+    } else if (!this.isEdit) {
+      this.postNewEntry();
+    } else {
+      this.editEntry.projectId = this.TimeEntryForm.get('projectID').value;
+      this.editEntry.projectTaskId = this.TimeEntryForm.get('taskID').value;
+      this.editEntry.durationInMin = this.TimeEntryForm.get('hours').value * 60 + this.TimeEntryForm.get('minutes').value;
+      this.editEntry.note = this.TimeEntryForm.get('notes').value;
+      this.updateEntry();
     }
-
     this.closeDialog();
```

Comment:
what if error happened on `updateEntry` call... since it's `async`, you will not give user a chance to correct it's mistakes